adapters/repos/db/vector_index.go

```diff
@@ -42,4 +43,5 @@ type VectorIndex interface {
 	ContainsNode(id uint64) bool
 	AlreadyIndexed() uint64
 	DistancerProvider() distancer.Provider
+	NewQueryVectorDistancer(queryVector []float32) common.QueryVectorDistancer
```

Comment:
I would rename this method to `GetQueryVectorDistancer` or just `QueryVectorDistancer`